GNUmakefile

```diff
@@ -160,6 +160,11 @@ test: other-consul dev-build vet
 	@# _something_ to stop them terminating us due to inactivity...
 	{ go test $(GOTEST_FLAGS) -tags '$(GOTAGS)' -timeout 7m $(GOTEST_PKGS) 2>&1 ; echo $$? > exit-code ; } | tee test.log | egrep '^(ok|FAIL)\s*github.com/hashicorp/consul'
 	@echo "Exit code: $$(cat exit-code)" >> test.log
+	@if [ "0" != "$$(cat exit-code)" ]; then \
+	  echo Retrying to avoid flacky tests >> test.log;\
+	  echo Retrying tests once...;\
+	  go test -p 5 -parallel 1 -tags '$(GOTAGS)' -timeout 5m $(GOTEST_PKGS) 2>&1; echo $$? > exit-code;\
```

Comment:
Is there a reason we hard-code `-p 5 -parallel 1` here instead of using the value set in the CI script?

It's reasonable that on a second pass you have only a few failures in separate packages to parallelizing packages but not tests in a package makes sense. There is a risk with -parallel 1 that a failure in one of the big packages will cause it to timeout but I suppose this is better than what we have now anyway.

I'll merge it as-is and we will have some proper attention on making the test suite solid really soon!

Reply:
@banks 
* `-p 5` => will pass faster on different packages (since we expect only a few of them not to pass the first time) => avoid to be likely killed by timeout on the Travis side
* `-parallel 1` => will avoid parallelisation to be sure tests failing due to parallelisation won't fail this time

This is completely empiric (I did test a few configurations, this one was the first to work)